src/sentry/web/frontend/base.py

```diff
@@ -378,6 +372,12 @@ def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpRespo
             # In customer domain contexts, we will need to pre-populate the organization_slug keyword argument.
             kwargs["organization_slug"] = organization_slug
 
+        if self.is_auth_required(request, *args, **kwargs):
```

Comment:
can you add a unit test that recreates the behavior? (if we've run into the bug once, we'll probably run into it again)